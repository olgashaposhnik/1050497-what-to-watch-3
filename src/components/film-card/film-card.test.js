import React from "react";
import renderer from "react-test-renderer";
import FilmCard from "./film-card.jsx";

const film = {
  image: `img/what-we-do-in-the-shadows.jpg`,
  title: `What We Do in the Shadows`,
  genre: `comedy`,
  fullImage: `https://www.google.com/url?sa=i&url=https%3A%2F%2Ftsn.ua%2Fru%2Fbooks%2Fv-seti-poyavilsya-tizer-k-filmu-zahar-berkut-video-1358730.html&psig=AOvVaw0KWyl2zdo6yLT0CN3BJlnw&ust=1583063472147000&source=images&cd=vfe&ved=0CAIQjRxqFwoTCLDB2JfZ9ucCFQAAAAAdAAAAABAD`,
  director: `Peter Jackson`,
  starring: [
    `Alan Alda`,
    `Antony Alda`,
    `Robert Alda`,
    `Norman Alden`,
    `Tom Aldredge`
  ],
  duration: `1h 04m`,
  year: 2011,
  preview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  rating: 3.8,
  ratingCount: 854,
  description: `Directed by Jennifer Lee and Chris Buck, this sequel to the family animated adventure Frozen carries on the adventures of the Snow Queen Elsa (Idina Menzel), Elsaâ€™s kind-hearted and optimistic sister Anna (Kristen Bell), the comedic snowman Olaf (Josh Gad) and mountain guru Kristoff (Jonathan Groff) as they venture deep into the forest to discover the truth about an ancient and legendary mystery of the kingdom they call home. Produced by Peter Del Vecho.`,
};

it(`Should render MovieCard component`, () => {
  const tree = renderer
    .create(
        <FilmCard
          film={film}
          onFilmCardClick = {() => {}}
          onFilmCardMouseOver = {() => {}}
          onFilmCardMouseLeave = {() => {}}
          isPlaying={true}
        />
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
