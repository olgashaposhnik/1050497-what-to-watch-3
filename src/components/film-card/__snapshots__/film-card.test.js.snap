// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render MovieCard component 1`] = `
<article
  className="small-movie-card catalog__movies-card"
  onClick={[Function]}
  onMouseOut={[Function]}
  onMouseOver={[Function]}
>
  <div
    className="small-movie-card__image"
  >
    <video
      autoPlay={true}
      controls={true}
      muted={true}
      onClick={[Function]}
      poster="img/what-we-do-in-the-shadows.jpg"
      width="100%"
    >
      <source
        src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
      />
    </video>
  </div>
  <h3
    className="small-movie-card__title"
  >
    <a
      className="small-movie-card__link"
      href="#"
    >
      What We Do in the Shadows
    </a>
  </h3>
</article>
`;
